### PR TITLE
unix: allow nesting of kqueue fds in uv_poll_start

### PR DESCRIPTION
**Pull request by bnoordhuis**
kqueue file descriptors don't support ioctl(FIONBIO) (or any other ioctl
for that matter) so retry using fcntl(F_GETFL) + fcntl(F_SETFL) when we
receive a ENOTTY error.

Fixes: https://github.com/libuv/libuv/issues/883
CI: https://ci.nodejs.org/job/libuv-test-commit/51/

R=@saghul?

### CODE REVIEW

File: test/test-poll.c [hunk 1]
```diff
@@ -601,3 +611,47 @@ TEST_IMPL(poll_bad_fdtype) {
...
+#ifdef __linux__
+TEST_IMPL(poll_nested_epoll) {
+  uv_poll_t poll_handle;
+  int fd;
+
+  fd = epoll_create(1);
+  ASSERT(fd != -1);
+
+  ASSERT(0 == uv_poll_init(uv_default_loop(), &poll_handle, fd));
+  ASSERT(0 == uv_poll_start(&poll_handle, UV_READABLE, (uv_poll_cb) abort));
```
Comment (saghul) [inline test/test-poll.c]: does this compile without a warning?

**Comment by saghul** [comment]
LGTM

File: test/test-poll.c (hunk 1 above)
Comment (bnoordhuis) [inline test/test-poll.c]: You mean the cast?  Yes.

Comment (saghul) [inline test/test-poll.c]: Yep, I meant the cast. TIL. LGTM, go for it!

**Comment by bnoordhuis** [comment]
Small fix-up for Solaris, I forgot to include `<sys/ioctl.h>`: https://ci.nodejs.org/job/libuv-test-commit/53/

**Comment by saghul** [comment]
Looks like it's still failing:

~~~~
src/unix/core.c: In function 'uv__nonblock_fast':
src/unix/core.c:521:19: error: 'FIONBIO' undeclared (first use in this function)
     r = ioctl(fd, FIONBIO, &set);
                   ^
src/unix/core.c:521:19: note: each undeclared identifier is reported only once for each function it appears in
src/unix/core.c: In function 'uv__cloexec_fast':
src/unix/core.c:535:25: error: 'FIOCLEX' undeclared (first use in this function)
     r = ioctl(fd, set ? FIOCLEX : FIONCLEX);
                         ^
src/unix/core.c:535:35: error: 'FIONCLEX' undeclared (first use in this function)
     r = ioctl(fd, set ? FIOCLEX : FIONCLEX);
~~~~

**Comment by bnoordhuis** [comment]
Okay, one more fix-up: https://ci.nodejs.org/job/libuv-test-commit/54/

**Comment by saghul** [comment]
Solaris is fixed now, LGTM!

**Comment by bnoordhuis** [comment]
Landed in c5c419f, thanks for the review.

I snuck in a small rename of `uv__nonblock_slow/uv__nonblock_fast` to `uv__nonblock_fcntl/uv__nonblock_ioctl` and likewise for the cloexec functions.  I think they're more clear and honest about what the functions do.

**Comment by saghul** [comment]
👍

**Pull request closed without merging**